### DISCUSSION TITLE
Is there a standard URI endpoint for checking status, i.e if an instance is up and functioning normally?

### DISCUSSION THREAD

**Discussion by alydevs** [dani-garcia/vaultwarden#5117] [Q&A] [answered]
I'm setting up a status page for my self-hosted services and would like to find a URL for each service that will work if a site is functioning normally. I didn't find any previous discussion specifically about a status endpoint. Internally there's a health check used for Docker etc of course, but the status page would ideally not require privileged access.
My site's login page sends a GET to `/api/organizations/undefined/policies/token?token=undefined&email=undefined&organizationUserId=undefined`, which returns HTTP 200 and `{}`; this would be suitable, I just wanted to check if this is stable and could be used to confirm a Vaultwarden instance is up and functioning normally. Thanks!

**Comment by jjlin** [comment, marked as answer]
The healthcheck API is not authenticated. You can use either `/alive` (the original Vaultwarden API) or `/api/alive` (the same API, just made available under a different path that Bitwarden added later on).

**Reply by alydevs** [reply]
Thank you! Is this documented somewhere I missed?

**Reply by jjlin** [reply]
Not really. There is not any official or even maintained documentation for the Bitwarden private/internal APIs AFAIK.

**Reply by alydevs** [reply]
That's fine. Thanks again for providing that info, I'm sure it'll help someone else as well in the future.

**Reply by bilalelhaj** [reply]
Just a heads-up: The image ships with a built-in `./healthcheck.sh` script. It's generally recommended to use that instead of a custom curl command, as it handles the check logic correctly/more robustly
